src/app/archive/lib/processor.ml

```diff
@@ -38,39 +38,89 @@ let txn_hash_to_base58_check ?(v1_transaction_hash = false) hash =
   if v1_transaction_hash then Transaction_hash.to_base58_check_v1 hash
   else Transaction_hash.to_base58_check hash
 
+let unwrap t = Result.map_error ~f:Caqti_error.show t |> Result.ok_or_failwith
+
+let ensure_local_copies (module Conn : Mina_caqti.CONNECTION) ~default tbl =
+  Hashtbl.find_or_add ~default tbl (Uri.to_string Conn.source)
+
+let load_copy' ~default ~local_copies ~typ ~query ~load_elt
+    (module Conn : Mina_caqti.CONNECTION) =
+  match Hashtbl.find local_copies (Uri.to_string Conn.source) with
+  | Some copy ->
+      Deferred.return copy
+  | None ->
+      let t_to_id = ensure_local_copies (module Conn) ~default local_copies in
+      let%bind all_rows =
+        Conn.collect_list (Mina_caqti.collect_req Caqti_type.unit typ query) ()
+      in
+      let%map () =
+        Deferred.List.iter (unwrap all_rows) ~f:(fun row ->
+            load_elt t_to_id row )
+      in
+      t_to_id
```

Comment:
If I'm reading this (and its usages) correctly, doesn't this end up sending most / all of the database over the network, and then holding it all in memory for the lifetime of the archive process?

If we imagine that the ledger is full with the maximum possible `2^35` accounts with distinct public keys, each public key can be optimally compressed into `32 = 2^5` bytes, so this results in sending / holding in RAM `2^40` bytes of data, or roughly 1TB.

Reply:
Good catch, I'm removing this optimization